### PR TITLE
HTCONDOR-861 Implement Miron's activation metrics

### PR DESCRIPTION
**Pull request by Todd-L-Miller**
Implement Miron's activation metrics.

# HTCondor Pull Request Checklist for internal reviewers

- [x] Verify that (GitHub thinks) the merge is clean. If it isn't, and you're confident you can resolve the conflicts, do so. Otherwise, send it back to the original developer.
- [x] Verify that the related Jira ticket exists and has a target version number and that it is correct.
- [x] Verify that the Jira ticket is in review status and is assigned to the reviewer.
- [x] Verify that the Jira ticket (HTCONDOR-xxx) is mentioned at the beginning of the title. Edit it, if not
- [x] Verify that the branch destination of the PR matches the target version of the ticket
- [x] Check for correctness of change
- [x] Check for regression test(s) of new features and bugfixes (if the feature doesn't require root)
- [x] Check for documentation, if needed
- [x] Check for version history, if needed
- [x] Check BaTLab dashboard for successful build (https://batlab.chtc.wisc.edu/results/workspace.php) and test for either the PR or a workspace build by the developer that has the Jira ticket as a comment.
- [x] Check that commit message references the Jira ticket (HTCONDOR-xxx)

## After the above
- Hit the merge button if the pull request is approved and it is not a security patch (security changes require 2 additional reviews)
- If the pull request is approved, take the ticket out of review state
- Assign JIRA Ticket back to the developer

### CODE REVIEW

**Review by markcoatsworth — CHANGES_REQUESTED** [review]
This code looks good. I haven't left many comments because overall I think this is very well done: code is clean, easy to understand, consistent with surrounding methods, well commented, lots of safety checks and has good tests. There's only one small bug that needs to get resolved before we merge this.

File: src/condor_tests/test_activation_metrics.py
Comment (Todd-L-Miller) [reply src/condor_tests/test_activation_metrics.py:79]: At some point while writing this test, I thought I would need `job_one_handle` for a purpose other than generating `job_one_ad`.  It's easy enough to merge the functions, if you think that'd be better.

File: src/condor_tests/cmd_q_protocols.run
Comment (Todd-L-Miller) [reply src/condor_tests/cmd_q_protocols.run:70]: See reply below.

File: src/condor_tests/cmd_q_protocols.run
Comment (Todd-L-Miller) [reply src/condor_tests/cmd_q_protocols.run:213]: As far as it goes, I wanted to leave it in there but maybe should have added a comment about why: because turning off SHADOW_LAZY_QUEUE_UPDATE breaks the test, which means it way less robust than it at first may appear.

File: src/condor_starter.V6.1/user_proc.cpp
Comment (Todd-L-Miller) [reply src/condor_starter.V6.1/user_proc.cpp:156]: Derp!  Thanks.

File: src/condor_shadow.V6.1/remoteresource.cpp
Comment (Todd-L-Miller) [reply src/condor_shadow.V6.1/remoteresource.cpp:320]: I spoke with ToddT, and his feeling is that "job exit" should cover "job was killed by being put on hold," so I have some work to do here,

Comment (Todd-L-Miller) [reply src/condor_shadow.V6.1/remoteresource.cpp]: I found a new place in the shadow to hang this code that seems to work.

File: src/condor_tests/cmd_q_protocols.run
Comment (JaimeFrey) [reply src/condor_tests/cmd_q_protocols.run:213]: A comment would be useful, but not required for this ticket.

File: src/condor_shadow.V6.1/remoteresource.cpp
Comment (JaimeFrey) [reply src/condor_shadow.V6.1/remoteresource.cpp]: The change looks good to me.

File: src/condor_tests/test_activation_metrics.py
Comment (JaimeFrey) [reply src/condor_tests/test_activation_metrics.py:79]: Merging doesn't seem necessary.

**Review by JaimeFrey — APPROVED** [review]
Todd has addressed Mark's comments for code review.

**Pull request merged**
